### DISCUSSION TITLE
Emitting events from a server in a sidecar?

### DISCUSSION THREAD

**Discussion by nathanielastudillo** [tauri-apps/tauri#4492] [Q&A]
Hey!

I have a clojure web server running in a sidecar that I'd like to listen to an event from. Something like:

1. Call endpoint on server (which is in a sidecar)
2. Server handler for that endpoint emits an event with some data (via. tauri's IPC?)
3. JS frontend listens to that event and does something

Step 2 is the point I'm stuck on. I can't figure out how to emit a message. How would I go about getting an event + data out of a sidecar?

**Comment by simonhyll** [comment]
I guess the easiest option is probably setting up something like a WebSocket between the app and the server where you can basically have a cross-binaries event bus. Other than that, afaik, the sidecar is ran as its own process entirely and would be very, very, very hard to communicate with directly (literally the same thing as just running two binaries on your computer and trying to smash them together 😅). It's not added as some kind of shared library or anything, it literally just baked into the rest of the app and executed as-is.
